### ISSUE TITLE
Entering title into Post field doesn't find the correct post

### ISSUE THREAD

**Issue by kienstra** [getblocklab/block-lab#430]
### Update: Please see [this comment](https://github.com/getblocklab/block-lab/issues/430#issuecomment-531337948). This search is behaving how you'd expect WordPress search to, but it'd be best if this somehow only searched page titles, not also their content.

Here's a [screencast](https://cloudup.com/ceXPaOvEgcN) of the issue the user provided. I  cropped it to hide the URL, and added some text in it.

**Steps To Reproduce**
(Though I couldn't reproduce this)
1. Create a Block Lab block with a Post field 
2. Set it to display pages (not posts)
3. Create a new post, and add that block to it
4. In the Post field, enter the name of a page
5. Expected: the page appears
6. Actual: the page didn't appear:

<img width="1179" alt="post-field" src="https://user-images.githubusercontent.com/4063887/64837707-6877f680-d5b5-11e9-8d21-436e0c012b83.png">

In the [screencast](https://cloudup.com/ceXPaOvEgcN), the requests for pages seem to take a long time.

Though in my local, the requests for pages complete quickly:

![requests-pages](https://user-images.githubusercontent.com/4063887/64837598-e1c31980-d5b4-11e9-890e-5ec9ab01e71d.gif)

I'm not sure what could cause this, but I'll look at it.

**Comment by lukecarbis** [comment]
I think I know what causes this. It's come up before. The WordPress search function includes both the title _and_ the post content when conducting a search. So in this example, any page which contains the text "Admissions" in the post content will show up too.

In terms of speed, here's a gif of how well it work for me running live. If the staging server this was running on is slow anyway, then the speed experienced in the screencast seems about right.

![2019-09-13 15-04-48 2019-09-13 15_05_55](https://user-images.githubusercontent.com/1097667/64838778-0f04db80-d638-11e9-8549-51a82903b724.gif)

I think we can close this.

**Comment by kienstra** [comment]
Hi @lukecarbis,
Ah, great point. It's searching the content also.

Like you mentioned, searching for simply 'Admissions' might not return the right title, as 'Admissions' could be in the content of a lot of posts.

Though it should probably still return the correct page when typing the entire title, right?

The user reported that this issue exists even when typing the entire title.

**Comment by kienstra** [comment]
Ah, maybe 'Admissions' was the page title. 

In that case, with the current way WordPress search works, we wouldn't expect it to find that title if 'Admissions' is in the content of enough pages.

**Comment by kienstra** [comment]
'Admissions' was the page title. So the search is currently behaving how we'd expect WordPress search to behave.

Still, maybe there's a way to limit the search to only page titles. May by passing an extra parameter to [apiFetch](https://github.com/getblocklab/block-lab/blob/41fd3922f44ce1dbb1c4677d5b7f54317c5959a6/js/blocks/components/fetch-input.js#L86), or filtering `pre_get_posts`.

**Comment by lukecarbis** [comment]
Some time searching didn't reveal any quick or easy way of doing this. There's no additional arguments we can send to the WP REST API endpoint to only search titles. We'd have to create a custom endpoint, and construct a custom database query.

I'd say that it's not worth including that extra code just to fix this one issue.

**Comment by kienstra** [comment]
Yeah, this wouldn't be easy. Like you mentioned, it'd require some custom work with the REST API, probably with a custom SQL query.

**Comment by kienstra** [comment]
Another user reported this issue as well.

**Comment by lukecarbis** [comment]
One option I just thought of – we could filter the query's "ORDERBY" parameter with the `[posts_orderby](http://codex.wordpress.org/Plugin_API/Filter_Reference/posts_orderby)` filter, and do a MYSQL order by `CASE` or `LOCATE`.

Essentially combining [this](https://wordpress.stackexchange.com/a/41723/33195) and [this](https://stackoverflow.com/a/18726122/1372282).

**Comment by kienstra** [comment]
Ah, interesting

**Comment by martinbon** [comment]
I do believe I have the same/similar issue occuring.

It suggests posts nicely at first
![blocklab-certificate-title](https://user-images.githubusercontent.com/34155711/67568027-c3b01380-f733-11e9-8ac1-91bf2f80ef58.jpg)

But then when I type the suggested post type in there manually, it can't find it.
![blocklab-certificate-title-invalid](https://user-images.githubusercontent.com/34155711/67568064-d296c600-f733-11e9-97a0-ffcd0a5585ce.jpg)

So it seems like it's totally ignoring the post title.

**Comment by kienstra** [comment]
@martinbon,
Thanks for bringing this up, and for your good screenshots. We've talked about prioritizing this. 

You can check back here to see the status, a PR that fixes this will link to this issue.
